controllers/v1/observationsController.js

```diff
@@ -631,128 +630,140 @@ module.exports = class Observations extends Abstract {
-                            );
+                        )
+                        
 
-                            response["message"] = messageData;
+                        response["message"] = messageData;
 
-                            response.result.push({
-                                "count" : 1,
-                                "data" : data
-                            });
-
-                        } else {
-                            response["message"] = 
-                            messageConstants.apiResponses.ENTITY_NOT_FOUND;
-                            
-                            response.result.push({
-                                "count":0,
-                                "data" : []
-                            });
-                        }  
-
-                        return resolve(response);
+                        response.result.push({
+                            "data" : data,
+                            "count" : totalcount
+                        });
+                        
                     }
-                }
+                    return resolve(response)
 
-                let userAclInformation = await userExtensionHelper.userAccessControlList(
-                    userId
-                );
+                    
+                    
+                }
+            //++++++++important- check with Akash --- removed because db doesn't contain data.And learners api response data can't be queried even if data exist in DB
+                // let userAclInformation = await userExtensionHelper.userAccessControlList(
+                //     userId
+                // );
 
-                let tags = [];
+                // let tags = [];
                 
-                if( 
-                    userAclInformation.success && 
-                    Object.keys(userAclInformation.acl).length > 0 
-                ) {
-                    Object.values(userAclInformation.acl).forEach(acl=>{
-                        tags = tags.concat(acl);
-                    })
-                }
+                // if( 
+                //     userAclInformation.success && 
+                //     Object.keys(userAclInformation.acl).length > 0 
+                // ) {
+                //     Object.values(userAclInformation.acl).forEach(acl=>{
+                //         tags = tags.concat(acl);
+                //     })
+                // }
 
                 let entityDocuments = await entitiesHelper.search(
-                    result.entityTypeId, 
+                    result.entityType, 
                     req.searchText, 
                     req.pageSize, 
                     req.pageNo, 
                     
                     userAllowedEntities && userAllowedEntities.length > 0 ? 
                     userAllowedEntities : 
                     false,
-                    tags
+                    
                 );
-
+            
+                if ( !(entityDocuments[0].count) ) {
```

Comment:
@VISHNUDAS-tunerlabs cross-check with data also , check count + data in if condition

Reply:
this implementation will change once the org search problem sort out, for now skipping this api